### PR TITLE
CASMHMS-5427 Build HMNFD with GitHub Actions and convert to Helm CT tests

### PR DESCRIPTION
**Pull request by schooler-hpe**
### Summary and Scope

This PR includes the following changes:

* Update HMNFD to build using GitHub Actions instead of Jenkins
* Pull images from algol60 instead of arti.dev.cray.com
* Use HSM /v2 rather than /v1 which will soon be removed
* Update Alpine base image version
* Stop building CT test RPMs and instead package/execute them via Helm
* Add a runCT.sh script/workflow that stands up an instance of HMNFD and executes the CT tests against it
* Refactored runIntegration.sh and the disruptive Tavern integration tests

### Issues and Related PRs

* Resolves CASMHMS-5427.
* Resolves CASMHMS-5415.

### Testing

This change was tested by executing the existing HMNFD unit and integration tests and verifying that they continued to pass. The HMNFD CT tests were also run as part of the new GitHub Actions workflow and verified to pass. The new version of HMNFD was also installed on Mug using Helm and confirmed to succeed by executing the new CT test Helm job, as well as the former RPM version of the CT tests, and verified to pass. Logs were inspected as well and were found to only contain failures/errors attributable to an unhealthy system environment and not caused by these changes.

Was a fresh Install tested? N
Was an Upgrade tested? Y
Was a Downgrade tested? Y

### Risks and Mitigations

Moderate risk since this is a significant refactoring of how HMNFD builds and is tested, but the core functionality is unchanged.

### CODE REVIEW

**Review by mjendrysik-hpe — APPROVED** [review]

File: Test/ct/smoke/smoke.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
+  "smoke_test_name": "hmnfd-smoke-tests",
+  "default_base_url": "http://cray-hmnfd/hmi/",
```
Comment (nieuwsma) [inline Test/ct/smoke/smoke.json:3]: so is the path http://cray-hmnfd/hmi/hmi/v1/health?

File: Dockerfile.ct.testing [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# MIT License
```
Comment (nieuwsma) [inline Dockerfile.ct.testing:1]: why did you make this file? What is different between Dockerfile and Dockerfile.ct.testing?

if its just environment variables, they can be set in the docker compose file

**Review by nieuwsma — CHANGES_REQUESTED** [review]
I wonder if the 'fake' stuff in the api-testing directory can be replaced with real versions?

File: Test/ct/smoke/smoke.json (hunk 1 above)
Comment (schooler-hpe) [reply Test/ct/smoke/smoke.json]: No, only one hmi should be there. Fixed.

File: Dockerfile.ct.testing (hunk 2 above)
Comment (schooler-hpe) [reply Dockerfile.ct.testing]: Good point. I've deleted Dockerfile.ct.testing and am now using Dockerfile in docker-compose.test.ct.yaml.

**Comment by schooler-hpe** [comment]
> 

The integration tests have been refactored and no longer use the fake components.

**Pull request merged**